core/trino-spi/src/main/java/io/trino/spi/ptf/TableFunctionProcessorProvider.java

```diff
@@ -32,7 +33,7 @@ default TableFunctionDataProcessor getDataProcessor(ConnectorTableFunctionHandle
      * This method returns a {@code TableFunctionSplitProcessor}. All the necessary information collected during analysis is available
      * in the form of {@link ConnectorTableFunctionHandle}. It is called once per each split processed by the table function.
      */
-    default TableFunctionSplitProcessor getSplitProcessor(ConnectorSession session, ConnectorTableFunctionHandle handle)
+    default TableFunctionSplitProcessor getSplitProcessor(ConnectorSession session, ConnectorTableFunctionHandle handle, ConnectorSplit split)
```

Comment:
To me we could also return ConnectorPageSource here. If so, this would be a follow-up after this is agreed with @kasiafi @martint .